src/load.jl

```diff
@@ -24,7 +24,11 @@ function load(tf::TiffFile; verbose=true, mmap = false)
         nplanes += 1
     end
 
-    if mmap
+    ifd = first(ifds)
+    if mmap && iscontiguous(ifd) && getdata(CompressionType, ifd, COMPRESSION, COMPRESSION_NONE) === COMPRESSION_NONE
+        return MmappedTIFF(tf, ifds)
```

Comment:
> Let's wait for https://github.com/timholy/TiffImages.jl/pull/2 before merging this.

I now don't believe #2 is the correct fix. We should fix the semantics instead. 

Note the `close(tf.io)` in R42 -- this means `mmap=true` never correctly close the file IO -- the fact that linux CI passes the test is just some bad luck.

Instead, we should use the explicit `mmap` semantic in https://github.com/timholy/TiffImages.jl/pull/3 and it's a perfect fix for https://github.com/tlnagy/TiffImages.jl/pull/79/files#r880478304

```julia
img = rand(Gray, 64, 64)
TiffImages.save("tmp.tif", img)
sz = TiffImages.mmap_load("tmp.tif") do img
    size(img)
end
img2 = TiffImages.load("tmp.tif") # works
TiffImages.save("tmp.tif", img2) # works
```

---

We could even bring this `mmap` function to FileIO, although I'm not sure if it is feasible.

Reply:
>  the fact that linux CI passes the test is just some bad luck.

Not really, see https://stackoverflow.com/questions/42961339/mmap-after-deleting-the-file. I wish Windows did this. I think that's really the root of all the problems on Windows, and everything works nicely as you'd hope on Linux.